Bento/Renderable/Deletable.swift

```diff
@@ -15,3 +13,15 @@ public extension Deletable {
         return nil
     }
 }
+
+// NOTE: Workaround to the lack of generalized existentials for Protocols with
+//       Associated Types (PATs).
+//
+//       See `_Focusable` for the rationale.
+/// Bento internal protocol. See `Deletable` for the public API.
+public protocol _Deletable {
```

Comment:
Why is this changed? 
It was working fine before no?

Reply:
I want to keep `Focusable` inheriting from `Renderable` as `Deletable` does. However, with the existing configuration it would require two extra existential wrappers (`AnyFocusableBox` and `AnyFocusableDeletableBox`), and even more stuff being dumped into `AnyRenderable`. This isn't a scalable solution if we are going to introduce more protocols, where the # of existential wrappers grows exponentially.

Reply:
Why do we need it to `inherit` from `Renderable`?

Reply:
It isn't necessary for `Deletable`. But `Focusable` enforces a `View: FocusableView` constraint on the component.

Reply:
Alternatively, we can drop both `Focusable where Self.View: FocusableView` and `Deletable: Renderable` constraints. Then these protocols would have no associated type and therefore can be used as existentials.

Reply:
Sounds good 👍 
We can also get rid of `Deletable` inherit from `Renderable`